added Type Jam puzzle app for review

malloc-error opened the pull request:
Following on conversation with @domesticmouse this PR is to request a quality check/advice on my demo puzzle with variable fonts and shaders before publishing for a blog post.

Among other things, I'm sure there is an issue with the way I coded the stateless widgets, forcing me to use `WidgetsBinding.instance.addPostFrameCallback` in the build method of some widgets to force setState to trigger a rebuild. 

My puzzle 'data' model is also functioning but of questionable usefulness/quality.

## Pre-launch Checklist

- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I read the [Contributors Guide].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-devrel channel on [Discord].

<!-- Links -->
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Contributors Guide]: https://github.com/flutter/samples/blob/main/CONTRIBUTING.md

domesticmouse commented:
Please:
  - Update this PR to the latest from `main` to fix the CI issues
  - Update [tool/flutter_ci_script_beta.sh](https://github.com/flutter/samples/blob/main/tool/flutter_ci_script_beta.sh) and [tool/flutter_ci_script_master.sh](https://github.com/flutter/samples/blob/main/tool/flutter_ci_script_master.sh) to reference this project to include it in CI
  - Update [tool/flutter_ci_script_stable.sh](https://github.com/flutter/samples/blob/main/tool/flutter_ci_script_stable.sh) to reference this project, commted out with a TODO against me to include it back in when `stable` increments.

domesticmouse commented:
@malloc-error why the close?

malloc-error commented:
@domesticmouse Oh oops, I thought this was just to get a quick way to look at it to confirm it's good enough to show out, but it would live on my repo. I will reopen!

domesticmouse commented:
Please ping this thread when you have added a license screen

malloc-error commented:
@domesticmouse I have added a license screen widget, and manually added the relevant licenses for the two fonts. It shows up at the very end of the game. Also updated to latest from `main` and changed the relevant script files.

domesticmouse commented:
Looks like there are some analysis issues:

```console
Analyzing varfont_shader_puzzle...

   info - lib/components/fragment_shaded.dart:115:28 - The value of the field '_identity' isn't used. Try removing the field, or using it. - unused_field
   info - lib/components/rotator_puzzle.dart:166:22 - The local variable '_img' starts with an underscore. Try renaming the variable to not start with an underscore. - no_leading_underscores_for_local_identifiers
   info - lib/components/rotator_puzzle.dart:241:7 - This class (or a class that this class inherits from) is marked as '@immutable', but one or more of its instance fields aren't final: RotatorPuzzleTile.row, RotatorPuzzleTile.col - must_be_immutable
   info - lib/components/rotator_puzzle.dart:279:45 - Don't put any logic in 'createState'. Try moving the logic out of 'createState'. - no_logic_in_create_state
   info - lib/components/wonky_char.dart:156:15 - Don't invoke 'print' in production code. Try using a logging framework. - avoid_print
   info - lib/page_content/pages_flow.dart:102:38 - Don't put any logic in 'createState'. Try moving the logic out of 'createState'. - no_logic_in_create_state
   info - lib/page_content/pages_flow.dart:137:3 - Constructors in '@immutable' classes should be declared as 'const'. Try adding 'const' to the constructor declaration. - prefer_const_constructors_in_immutables
   info - lib/page_content/wallpapers_flow.dart:6:8 - Unused import: '../components/components.dart'. Try removing the import directive. - unused_import
```

domesticmouse commented on tool/flutter_ci_script_stable.sh:
```diff
@@ -22,6 +22,8 @@ declare -ar PROJECT_NAMES=(
     "experimental/web_dashboard"
     "experimental/linting_tool"
+    # TODO(DomesticMouse): include when stable increments
+    # "experimental/varfont_shader_puzzle"
```
What is it that precludes this sample from running on Flutter stable? If there is a feature in Flutter 3.7.0-beta, please mark up the `pubspec.yaml` as having a minimum supported Flutter version.

malloc-error replied:
`FragmentShader` and the `toImageSync()` method for `RepaintRenderBoundary` are only available in master and beta at the moment, so I've commented this project out of stable for now.

I've marked the current beta release as minimum in `pubspec.yaml` since the `FragmentShader` API changed somewhat recently, and my code will break under the older API, but I can't tell exactly which beta release that API change occurred in.

domesticmouse reviewed (changes requested):
There are some relatively minor issues that need mitigation, but apart from that this looks good to land. Please review and ping if you have questions or you are ready for another review pass.

malloc-error commented on experimental/varfont_shader_puzzle/lib/components/components.dart:
```diff
@@ -0,0 +1,10 @@
+
+export 'wonky_char.dart';
+export 'wonky_anim_palette.dart';
+export '../page_content/pages_flow.dart';
```
Yes, I was trying to have everything packed into that one components.dart file, but it makes more sense to import directly in each necessary file. I believe I've fixed it to how you are suggesting.

malloc-error commented on experimental/varfont_shader_puzzle/lib/components/fragment_shaded.dart:
```diff
@@ -0,0 +1,275 @@
+
+  @override
+  void paint(ui.Image image, Size size, ui.Canvas canvas) {
+    // final ImageShader sampler =
```
That was left over from the initial FragmentShader API. I've removed it now.

malloc-error commented on experimental/varfont_shader_puzzle/pubspec.yaml:
```diff
@@ -0,0 +1,107 @@
+version: 1.0.0+1
+
+environment:
+  sdk: '>=2.18.0-271.7.beta <3.0.0'
```
Updated accordingly

malloc-error commented on experimental/varfont_shader_puzzle/pubspec.yaml:
```diff
@@ -0,0 +1,107 @@
+  # Use with the CupertinoIcons class for iOS style icons.
+  cupertino_icons: ^1.0.2
+  google_fonts: ^3.0.1
+  # vector_math: ^2.1.3
```
Deleted the commented out dependencies

malloc-error commented on experimental/varfont_shader_puzzle/pubspec.yaml:
```diff
@@ -0,0 +1,107 @@
+name: type_puzzle_2
```
Changed name accordingly

malloc-error commented:
@domesticmouse Thank you for your feedback (and patience)! All checks are now showing green, and I believe I have addressed all the issues in your change request. I also made a few additional comments for clarity, or to add info about variable fonts.

Just flagging that there are now 10 new commits in addition to the 3 from my first attempt. If that's going to make the log overly messy, I can look into squashing them!

domesticmouse commented:
Don't worry about squashing, we do it on merge.

domesticmouse commented on experimental/varfont_shader_puzzle/pubspec.yaml:
```diff
@@ -94,14 +88,4 @@ flutter:
-  #
-  # For details regarding fonts from package dependencies,
-  # see https://flutter.dev/custom-fonts/#from-packages
+        - asset: assets/fonts/Amstelvar-Roman[GRAD,XOPQ,XTRA,YOPQ,YTAS,YTDE,YTFI,YTLC,YTUC,wdth,wght,opsz].ttf
```
Nit: please add trailing newline at end of file.

malloc-error commented:
@domesticmouse Thanks!
